testdata/p4_16_samples_outputs/parser-non-field-lhs-bmv2-first.p4

```diff
@@ -0,0 +1,62 @@
+#include <core.p4>
+#define V1MODEL_VERSION 20180101
+#include <v1model.p4>
+
+header test_h {
+    bit<8> field;
+}
+
+struct header_t {
+    test_h[4] hs;
+}
+
+struct metadata_t {
+    bit<2> hs_next_index;
+}
+
+parser TestParser(packet_in pkt, out header_t hdr, inout metadata_t meta, inout standard_metadata_t std_meta) {
+    state start {
+        meta.hs_next_index = 2w0;
+        hdr.hs[meta.hs_next_index].setValid();
+        hdr.hs[meta.hs_next_index] = (test_h){field = 8w0};
+        meta.hs_next_index = meta.hs_next_index + 2w1;
+        transition hs_1;
+    }
+    state hs_1 {
+        hdr.hs[meta.hs_next_index].setValid();
+        hdr.hs[meta.hs_next_index].field = 8w1;
+        meta.hs_next_index = meta.hs_next_index + 2w1;
+        transition hs_2;
+    }
+    state hs_2 {
+        hdr.hs[meta.hs_next_index].setValid();
+        hdr.hs[meta.hs_next_index].field = 8w0;
+        hdr.hs[meta.hs_next_index].field[3:0] = 4w2;
+        meta.hs_next_index = meta.hs_next_index + 2w1;
+        transition hs_3;
+    }
+    state hs_3 {
+        hdr.hs[hdr.hs[0].field + 8w3].setValid();
+        hdr.hs[hdr.hs[1].field + 8w2].field = 8w0;
+        hdr.hs[hdr.hs[2].field + 8w1].field[3:0] = 4w3;
```

Comment:
could we use a `lookahead` expression here for the RHS to make the test more comprehensive?

Reply:
Done. I've also identified a bug in the `ExpandLookahead` pass and fixed it.